### PR TITLE
Refactoring generate users to email

### PR DESCRIPTION
**Pull request by ghop02**
As I was going through the code, I restructured the existing functions a bit in a way that made slightly more sense to me - lmk if you like this approach!

### CODE REVIEW

File: scripts/vaccination_alerts/generate-users-to-email.ts [hunk 1]
```diff
@@ -5,10 +5,69 @@ import {
...
+  alertSubscriptions.forEach(emailLocationsDoc => {
+    const email = emailLocationsDoc.id;
+    const { locations } = emailLocationsDoc.data();
+
+    // Currently only alerting on states, so pull state fips codes from subscribed locations.
+    const subscribedStateFips = getStateFipsCodesSet(locations);
+
+    const userFipsToAlert = subscribedStateFips
+      .intersection(fipsCodesToAlert)
+      .value();
```
Comment (pnavarrc) [inline scripts/vaccination_alerts/generate-users-to-email.ts:34]: It feels a bit weird for me to break a `_.chain` sequence, consider returning an array from `getStateFipsCodesSet` so we can just use `_.intersection` here.

File: scripts/vaccination_alerts/generate-users-to-email.ts [hunk 2]
```diff
@@ -84,7 +108,14 @@ async function main() {
...
 if (require.main === module) {
-  main()
+  const argv = yargs.options({
+    input: {
+      default: DEFAULT_ALERTS_FILE_PATH,
+      description: 'Updated alerts input path.',
+    },
+  }).argv;
+
+  main(argv.input)
```
Comment (pnavarrc) [inline scripts/vaccination_alerts/generate-users-to-email.ts:118]: 💯  - I'm going to update my script to send test emails after we merge this!

File: scripts/vaccination_alerts/generate-vaccine-alerts-info.ts [hunk 3]
```diff
@@ -4,33 +4,58 @@ import {
...
   const firebaseInfo = await getFirebaseVaccinationInfo();
 
+  console.log('Getting snapshot of alerts to send');
   const vaccinationAlertUpdates = getUpdatedVaccinationInfo(
     cmsInfo,
     firebaseInfo,
   );
 
-  fs.writeFileSync(alertsFilePath, JSON.stringify(vaccinationAlertUpdates));
+  for (const fips of Object.keys(vaccinationAlertUpdates)) {
```
Comment (pnavarrc) [inline scripts/vaccination_alerts/generate-vaccine-alerts-info.ts:32]: nit: consider using `const [fips, current] of Object.entries(vaccinationAlertUpdates)`

File: scripts/vaccination_alerts/generate-vaccine-alerts-info.ts [hunk 4]
```diff
@@ -4,33 +4,58 @@ import {
...
+    path: {
+      default: DEFAULT_ALERTS_FILE_PATH,
+      description: 'Output path.',
+    },
+  }).argv;
+
+  main(argv.path)
     .then(() => {
-      console.log(`Done. Generated ${alertsFilePath}`);
+      console.log(`Done. Generated `);
```
Comment (pnavarrc) [inline scripts/vaccination_alerts/generate-vaccine-alerts-info.ts:58]: Maybe log the path of the file here?

**Review by pnavarrc — APPROVED** [review]
Added some comments, looks good!

File: scripts/vaccination_alerts/generate-users-to-email.ts [hunk 5]
```diff
@@ -84,7 +108,14 @@ async function main() {
 }
 
 if (require.main === module) {
-  main()
+  const argv = yargs.options({
```
Comment (pnavarrc) [inline scripts/vaccination_alerts/generate-users-to-email.ts:111]: I think we need to add yargs to the dependencies of the project

File: scripts/vaccination_alerts/generate-users-to-email.ts (hunk 2 above)
Comment (pnavarrc) [reply scripts/vaccination_alerts/generate-users-to-email.ts:118]: missing semicolon (not that I care, but maybe Prettier is not running for you?)

File: scripts/vaccination_alerts/generate-users-to-email.ts (hunk 5 above)
Comment (ghop02) [reply scripts/vaccination_alerts/generate-users-to-email.ts:111]: yeah I think it's pulled in as a dependency but definitely should be explicitly defined - will do

File: scripts/vaccination_alerts/generate-users-to-email.ts (hunk 2 above)
Comment (ghop02) [reply scripts/vaccination_alerts/generate-users-to-email.ts:118]: I don't think there's a semicolon there because it's a chained operation

**Pull request merged**